tiledb/sm/query/query_condition.cc

```diff
@@ -1041,7 +1188,7 @@ void QueryCondition::apply_ast_node_dense(
     const uint64_t* buffer_offsets =
         static_cast<uint64_t*>(tile_offsets.data()) + src_cell;
     const uint64_t buffer_offsets_el =
-        tile_offsets.size() / constants::cell_var_offset_size;
+        (tile_offsets.size() / constants::cell_var_offset_size) - src_cell;
```

Comment:
This will work, but buffer_offsets and buffer_offsets_el names are not accurate. We could remove src_cell from buffer_offsets, buffer_offsets_el and buffer_validity... And then the first line of the loop could be src_buffer_index = start + src_cell + c * stride;

Reply:
Like such? https://github.com/TileDB-Inc/TileDB/pull/3766/commits/6a0115ac2416984de53f1d182cd2f5746a72c63e